mediapipe/tasks/ios/genai/inference/sources/LlmInference+Session.swift

```diff
@@ -0,0 +1,269 @@
+// Copyright 2024 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import MediaPipeTasksGenAIC
+
+extension LlmInference {
+
+  /// An `LlmInference` Session that can be used to execute queries using the LLM that was used to
+  /// initialize the `LlmInference`task.
+  /// You can create multiple query sessions using the same `LlmInference` which can be active at
+  /// the same time.
+  /// You can also clone an existing session and continue querying the LLM from where you left off.
+  ///
+  /// Note: Inherits from `NSObject` for Objective C interoperability.
+  @objc(MPPLLMInferenceSession) public final class Session: NSObject {
+    // Session runner that manages the creation, deletion and execution of the underlying C session.
+    private let llmSessionRunner: LlmSessionRunner
+
+    // LLM Inference used to create this session.
+    private let llmInference: LlmInference
```

Comment:
I am storing a strong reference to `llmInference` to ensure that any given session does not out live the `llmInference` and create all sorts of undefined behaviour. This won't cause a retain cycle since llmInference does not store a reference to any session.

Reply:
Thanks! Yeah agree that we need a strong reference to `llmInference` here to manage life cycles. 

> Also lets me synchronise the response generation calls across sessions to fix a crash because of simultaneous response generation calls.

I believe only 1 response generation can happen at a time, is this what you refer to?

Reply:
yes. across sessions.